packages/manager/apps/vrack-services/src/utils/vs-utils.ts

```diff
@@ -157,6 +163,60 @@ export const useUpdateVrackServices = ({
   };
 };
 
+export const useDeleteVrackServices = ({
+  vrackServices,
+  onSuccess,
+  onError,
+}: {
+  vrackServices: string;
+  onSuccess?: () => void;
+  onError?: (result: ApiError) => void;
+}) => {
+  const [isErrorVisible, setIsErrorVisible] = React.useState(false);
+  const queryClient = useQueryClient();
+  const { data: servicesId, isError, error } = useQuery<
+    ApiResponse<number[]>,
+    ApiError
+  >({
+    queryKey: GetVrackServicesServiceIdQueryKey({ vrackServices }),
+    queryFn: () => GetVrackServicesServiceId({ vrackServices }),
+    enabled: !!vrackServices,
+  });
+
+  const {
+    mutate: deleteVs,
+    isError: isTerminateError,
+    error: terminateError,
+    isSuccess,
+  } = useMutation({
+    mutationFn: () =>
+      DeleteVrackServices({
+        serviceId: servicesId?.data[0],
+      }),
+    mutationKey: DeleteVrackServicesQueryKey(vrackServices),
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: DeleteVrackServicesQueryKey(vrackServices),
```

Comment:
You want to invalidate the listing query instead to make the app reload the data-grid if it is impacted

Reply:
The datagrid is not impacted on this first action for resiliation. It will be impacted only after the resiliation is validated while clicking on the email link